pkg/controllers/infra/cluster_controller.go

```diff
@@ -141,13 +184,121 @@ func (r *ClusterController) deleteCAPIClusterIfNeeded(ctx context.Context, infra
 }
 
 func (r *ClusterController) reconcile(ctx context.Context, infraCluster *infrav1.Cluster) (ctrl.Result, error) {
-	switch infraCluster.Spec.InfraType {
-	case infrav1.AWSClusterInfraType:
-		// TODO: implement AWS reconcile logic
-		return ctrl.Result{}, nil
-	default:
-		return ctrl.Result{}, errors.Errorf("unsupported infra type %s", infraCluster.Spec.InfraType)
+	// TODO: precheck
+	scope := scope.NewCluster(infraCluster)
+	provider := infraprovider.NewProvider(r.Client, scope)
+	if err := provider.Reconcile(ctx, infraCluster); err != nil {
+		conditions.MarkFalse(infraCluster, infrav1.InfrastructureProviderProvisionedCondition, infrav1.InfrastructureProviderProvisionFailedReason,
+			capiv1.ConditionSeverityError, err.Error())
+		return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile AWS Cluster %s/%s", infraCluster.Namespace, infraCluster.Name)
+	}
+	// check Cluster status
+	if !provider.IsInitialized(ctx) {
+		conditions.MarkFalse(infraCluster, infrav1.InfrastructureProviderProvisionedCondition, infrav1.InfrastructureProviderProvisionFailedReason,
```

Comment:
We canot set failed condition, because normally the control plane needed several minutes to startup.

We can update `IsInitialized`method to return underlying cluster `error`.  When cluster.status.Phase = failed, the failureReason would be set. We can return that, and only when it it in failed phase, we set the `InfrastructureProviderProvisionedCondition` false